common/transaction/src/lib.rs

```diff
@@ -32,7 +32,35 @@ pub type ExtractedFeeResult<M> =
     MultiValue2<OptionalValue<EsdtTokenPayment<M>>, ManagedVec<M, EsdtTokenPayment<M>>>;
 pub type OptionalValueTransferDataTuple<M> =
     OptionalValue<MultiValue3<GasLimit, ManagedBuffer<M>, ManagedVec<M, ManagedBuffer<M>>>>;
-pub type StakeMultiArg<M> = MultiValue2<TokenIdentifier<M>, BigUint<M>>;
+
+#[type_abi]
+#[derive(TopEncode, TopDecode, NestedEncode, NestedDecode, ManagedVecItem, Clone)]
+pub struct StakeArgs<M: ManagedTypeApi> {
+    pub token_id: TokenIdentifier<M>,
+    pub amount: BigUint<M>,
+}
+
+impl<M: ManagedTypeApi> StakeArgs<M> {
+    pub fn new(token_id: TokenIdentifier<M>, amount: BigUint<M>) -> Self {
+        StakeArgs { token_id, amount }
+    }
+
+    pub fn get_default() -> Self {
```

Comment:
This function is not ok. The `additional_stake_required` should be optional, not "default"